File: app/Models/Expense.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Facades\Storage;
use Illuminate\Support\Str;

class Expense extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'uuid',
        'user_id',
        'expense_category_id',
        'title',
        'amount',
        'expense_date',
        'payment_method',
        'receipt_image',
        'note',
        'currency',
        'metadata',
    ];

    protected function casts(): array
    {
        return [
            'amount' => 'decimal:2',
            'expense_date' => 'date',
            'metadata' => 'array',
            'deleted_at' => 'datetime',
        ];
    }

    protected static function booted()
    {
        static::creating(function ($expense) {
            if (empty($expense->uuid)) {
                $expense->uuid = (string) Str::uuid();
            }
        });
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function category()
    {
        return $this->belongsTo(ExpenseCategory::class, 'expense_category_id');
    }

    public function getReceiptImageUrlAttribute()
    {
        if (empty($this->receipt_image)) {
            return null;
        }

        return Storage::disk('public')->url($this->receipt_image);
    }
}

File: flutter_expense/lib/models/expense.dart
import 'expense_category.dart';

class Expense {
  final int id;
  final String uuid;
  final String title;
  final double amount;
  final String? expenseDate;
  final String? paymentMethod;
  final String? receiptImageUrl;
  final String? note;
  final String currency;
  final ExpenseCategory? category;
  final String? createdAt;

  const Expense({
    required this.id,
    required this.uuid,
    required this.title,
    required this.amount,
    this.expenseDate,
    this.paymentMethod,
    this.receiptImageUrl,
    this.note,
    this.currency = 'VND',
    this.category,
    this.createdAt,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'] as int,
      uuid: json['uuid'] as String,
      title: json['title'] as String,
      amount: (json['amount'] as num).toDouble(),
      expenseDate: json['expense_date'] as String?,
      paymentMethod: json['payment_method'] as String?,
      receiptImageUrl: json['receipt_image_url'] as String?,
      note: json['note'] as String?,
      currency: json['currency'] as String? ?? 'VND',
      category: json['category'] == null
          ? null
          : ExpenseCategory.fromJson(json['category'] as Map<String, dynamic>),
      createdAt: json['created_at'] as String?,
    );
  }
}